File: Chatbot-Studi.com/PublicSiteChatbot.Front/PoAssistant.Front/Client/ConversationRequestModel.cs
using Newtonsoft.Json;

namespace PoAssistant.Front.Client;

using System.Text.Json.Serialization;

public class ConversationRequestModel
{
    [JsonPropertyName("conversation")]
    public List<MessageRequestModel> Messages { get; set; } = new List<MessageRequestModel>();
}

public class MessageRequestModel
{
    [JsonPropertyName("role")]
    public string Role { get; set; } = "";

    [JsonPropertyName("content")]
    public string Content { get; set; } = "";

    [JsonPropertyName("duration_seconds")]
    public float DurationSeconds { get; set; } = 0.0f;
}

File: TwillioConnector/src/api_client/request_models/conversation_request_model.py
from uuid import UUID
from pydantic import BaseModel

class MessageRequestModel(BaseModel):
    role: str
    content: str
    duration_seconds: float = 0.0

    def to_dict(self):
        return {
            "role": self.role,
            "content": self.content,
            "duration_seconds": self.duration_seconds
        }

class ConversationRequestModel(BaseModel):
    user_id: UUID
    messages: list[MessageRequestModel]

    def to_dict(self):
        return {
            "user_id": str(self.user_id),
            "messages": [message.to_dict() for message in self.messages]
        }
